### ISSUE TITLE
Subword navigation : Apply introduced feature to whole UI

### ISSUE THREAD

**Issue by V-ed** [microsoft/vscode#53498]
Issue type : Feature Request

After the PR #48023 has been merged, a feature I really like has been added to VSCode and is currently in ~Insiders~ `1.26`. This change is working well so far in the editors, but I would want to be more extended through the whole vscode UI.

By "_through the whole UI_", I mean Search boxes, Command Palette, **file/folder renaming**, even the terminal (_stretched request here_).

This is a request that is already present in the extensions that provided this functionnality in the past (but were blocked by extensions APIs limits) : https://github.com/ow--/vscode-subword-navigation/issues/12 (as seen in this thread too : https://github.com/Microsoft/vscode/issues/541#issuecomment-371194836).

@kumarharsh's request (https://github.com/Microsoft/vscode/issues/3526#issuecomment-249657993) is basically the same as what this current issue is about.

It would be awesome to have either :
- an extension API to allow for hooks in the Input fields of VSCode (which would allow for something kind of like #35993) (**_this would be better in another issue though_**)
- or to actually do it in the core of vscode.

<details><summary>VSCode Insiders Info</summary>
<p>

```
  Version: 1.25.0-insider
  Commit: 33ba00346fd1bc889d5adc2f386c64fdb3657e61
  Date: 2018-07-03T14:40:32.630Z
  Electron: 1.7.12
  Chrome: 58.0.3029.110
  Node.js: 7.9.0
  V8: 5.8.283.38
  Architecture: x64
```

</p>
</details>

**Comment by davidcoleman007** [comment]
+1 million times yes, Please & Thank you!

**Comment by Tyriar** [comment]
Removing terminal label, I don't think this would be a good idea there (and isn't possible in many cases).

**Comment by roblourens** [comment]
I think it's currently not practical to implement this in the inputs that are currently native browser inputs (search, command palette, etc)

Doing that would require making those inputs full editors, which we do in a couple key places, but we've decided is too expensive to do everywhere.

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!

**Comment by sagemintblue** [comment]
+1 to @bj97301 's comment above.

As a new VS Code user coming from JetBrains editors like Intellij IDEA I was very confused by the implementation of subword navigation in VS Code. Adding subword navigation to input text fields e.g. for text search would be a great improvement in my opinion.

**Comment by bj97301** [comment]
> +1 to @bj97301 's comment above.
> 
> 
> 
> As a new VS Code user coming from JetBrains editors like Intellij IDEA I was very confused by the implementation of subword navigation in VS Code. Adding subword navigation to input text fields e.g. for text search would be a great improvement in my opinion.

Yes!  I am hoping they can compromise on doing sub word navigation.
